Rubberduck.SettingsProvider/XmlPersistanceService.cs

```diff
@@ -7,29 +7,27 @@
 
 namespace Rubberduck.SettingsProvider
 {
-    public class XmlPersistanceService<T> : XmlPersistanceServiceBase<T> where T : class, IEquatable<T>, new()
+    public class XmlPersistanceService<T> : XmlPersistanceServiceBase<T> 
+        where T : class, IEquatable<T>, new()
     {
-        public override T Load(T toDeserialize)
-        {
-            var defaultOutput = CachedOrNotFound();
-            if (defaultOutput != null)
-            {
-                return defaultOutput;
-            }
+        public XmlPersistanceService(IPersistancePathProvider pathProvider) : base(pathProvider) { }
 
-            var doc = GetConfigurationDoc(FilePath);
+        public override T Load(T toDeserialize, string nonDefaultFilePath = null)
+        {
+            var filePath = string.IsNullOrWhiteSpace(nonDefaultFilePath) ? FilePath : nonDefaultFilePath;
+            var doc = GetConfigurationDoc(filePath);
             var node = GetNodeByName(doc, typeof(T).Name);
             if (node == null)
             {
-                return FailedLoadReturnValue();
+                return Cached;
```

Comment:
This does not look as if it is doing the same as before the refactoring. Previously, the cached value was used if it exists and no file read was performed in that case. Now, the cached value is used only if the load from file fails.

Moreover, using the cached value without testing that something got cached in the first place is counterintuitive.

Reply:
The file read actually occurs in the `GetConfigurationDoc` call. This change does change the behavior, but it returns the cached settings object instead of deferring to the caller to obtain the defaults. The term "`Cached`" makes this confusing though, in that `Cached` will only be non-null if there had never been a successful read. If nothing has been cached, returning null is correct (and `Cached` would be null in that case). This covers the scenario where the settings file is deleted or broken at run-time, and changes the behavior to use the last successful load instead of the defaults in that case.

Reply:
Max might have a different thought, but maybe a comment?  I know exactly what's going on because I initially wrote the code and recently discussed it on chat, but the future me only wrote the code...